Weekly sync 230424

wenchenvincent opened the pull request:
https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/4263

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-04-19

PiperOrigin-RevId: 525384168

tensorflower-gardener committed:
Update GraphDef version to 1471.

PiperOrigin-RevId: 525384302

qukhan committed:
Cleanup TFLite subgraph.cc includes.

PiperOrigin-RevId: 525393164

fergushenderson committed:
Fix some incorrect references to "third_party/".

PiperOrigin-RevId: 525395399

fergushenderson committed:
Remove some incorrect references to "third_party/".

PiperOrigin-RevId: 525396094

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@98f5a340975b

Updates LLVM usage to match
[98f5a340975b](https://github.com/llvm/llvm-project/commit/98f5a340975b)

PiperOrigin-RevId: 525401578

tensorflower-gardener committed:
Merge pull request #60355 from fadara01:use_acl_instead_of_gemm_api

PiperOrigin-RevId: 525416639

jamesmullenbach committed:
Override safe_sparse_embedding_lookup for ShardedVariable input to ensure efficient sharded embeddings.

PiperOrigin-RevId: 525420722

tensorflower-gardener committed:
Use a custom rule for API files generation (fourth try)

Trying to submit https://github.com/tensorflow/tensorflow/pull/60180

PiperOrigin-RevId: 525437936

tensorflower-gardener committed:
Merge pull request #52576 from johan-gras:upstream/batchspaceint16

PiperOrigin-RevId: 525449619

Commit:
[XLA:GPU][NFC] Add the missing device memory size field to the test device info.

Also improve the device info test to make it harder to forget fields in the future.

PiperOrigin-RevId: 525462388

elfringham committed:
[Linaro:ARM_CI] Clean the extended skip list

Remove tests that pass from the skip list for ARM CI Extended

tensorflower-gardener committed:
Merge pull request #60359 from Intel-tensorflow:bhavanis/onednn-3.0-softmax-fwd

PiperOrigin-RevId: 525469474

Commit:
[XLA:GPU][NFC] Use common GPU device info in tests.

As a consequence fix fused_scatter test wrongly expecting rematerialization due to previously assumed zero device memory size.

PiperOrigin-RevId: 525470797

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525470850

mikedelorimier committed:
TPUCompileMlirPlaceholderProgramKey can be above the parallel_execute. This is used by map_outside_compilation.

PiperOrigin-RevId: 525474141

AspirinSJL committed:
Record op cost in CPU cycles

PiperOrigin-RevId: 525474427

Commit:
Adopt opaque pointers in kernel_gen.

PiperOrigin-RevId: 525477999

bhavani-subramanian committed:
Addressed review comments for _MklQuantizeV2 op

trisolaran committed:
more nway merge change for the deref_view.

PiperOrigin-RevId: 525486470

angerson committed:
[create-pull-request] automated change

faizan-m committed:
Access cancellation_context only once

PiperOrigin-RevId: 525487179

tensorflower-gardener committed:
Improve error reporting in TF runtime when tensors exceed RPC size limits. Having the shape information gives us more bread crumbs when trying to figure out which tensor caused this.

PiperOrigin-RevId: 525487478

BrianWieder committed:
Create new Windows RBE toolchain

PiperOrigin-RevId: 525489441

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525490459

rjpower committed:
Add explicit functions for testing for Graph/Eager tensor types.

This helps us transition to different internal types for Graph vs Eager tensors
without breaking our external interface.

PiperOrigin-RevId: 525491294

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525494104

BrianWieder committed:
Update references to `python/platform:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525497320

terryheo committed:
Add a test model of intermediate_tensor_output

This model shares output with its intermediate tensor.

PiperOrigin-RevId: 525508554

nkovela1 committed:
Adds TFLite conversion tests for the new `model.export()` API in Keras.

PiperOrigin-RevId: 525511900

qlzh727 committed:
Fix a divide_by_zero error in the tf.nn.weighted_moments().

The existing implementation will return NAN since it uses tf.math.reciprocal (which convert 0 to inf), and then result into inf/inf.

Change the implementation to div_no_nan() to return as 0 when weights are all 0s.

PiperOrigin-RevId: 525515850

tensorflower-gardener committed:
Improves compilation time by:

1. using InlinedVector.
2. avoiding making copies, and returning early from a loop.
3. reordering conjuncts by moving the expensive one as the last conjunct as opposed to eagerly evaluating it.

PiperOrigin-RevId: 525520523

tensorflower-gardener committed:
Merge pull request #60303 from kushanam:enable_binary_kernels_batch1

PiperOrigin-RevId: 525528433

GleasonK committed:
Integrate StableHLO at openxla/stablehlo@e6427c

PiperOrigin-RevId: 525539500

tensorflower-gardener committed:
Changing a few inline tensorflow methods to reduce binary sizes

PiperOrigin-RevId: 525547908

kanglant committed:
Switch to using Clang as the default compiler for TF Linux CI builds

PiperOrigin-RevId: 525556673

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525557978

BrianWieder committed:
Replace TF's `gen_api_init_files` dependency on `tensorflow/python:util` with the fine-grained targets.

PiperOrigin-RevId: 525562767

olegshyshkov committed:
[XLA:CPU Next] Add inits operands to gml_st.fusion.

gml_st.fusion implements DPS interface. It worked on tensor, because we can know the number of inits from the number of results. Now, we need gml_st.fusion to also work on memref, so we need an explicit information about the number of init arguments.

PiperOrigin-RevId: 525562793

rjpower committed:
Rename grpc_response_cache to make clear it is not gRPC specific.

Add logging and metrics for when the cache is used.

PiperOrigin-RevId: 525565931

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525583460

zycdragonball committed:
Allow ShardingParam to be used as attr

PiperOrigin-RevId: 525589128

rjpower committed:
Adjust checks for `type(Tensor)` to isinstance or is_eager/is_symbolic_tensor.

PiperOrigin-RevId: 525592807

BrianWieder committed:
Create new Windows RBE toolchain

PiperOrigin-RevId: 525598497

fionalang committed:
Deprecate `//third_party/tensorflow/python/types:types` and create fine-grained targets to replace it.

Once all references have been moved away from the large target, it can be safely deleted.

PiperOrigin-RevId: 525603943

terryheo committed:
gpu_delegate: Fix a bug when output tensors are shared among nodes

GraphFloat32's outputs is a list of values without consumers.
When a output tensor is consumed by another nodes, it's not listed properly.
This CL resolves it by maintaining `known_graph_outputs_` list.

Also GpuModel has a similar issue and fixed it.

PiperOrigin-RevId: 525604255

sparc1998 committed:
Refactor the MSA heap simulator code that creates a free chunks map into its own function. We will call this method multiple times with different slice dimensions in the upcoming changes to support sliced prefetching.

PiperOrigin-RevId: 525604318

hhb committed:
Remove rendezvous_creator_ from context

This unifies the worker and master code paths.

PiperOrigin-RevId: 525605840

mahmoud-abuzaina committed:
Changed DNNL_ARG_DST to DNNL_ARG_SRC in Reorder

pschuh committed:
Improve testing for custom_partitioning.

Add a test to demonstrate how to force XLA to choose
a different sharding.

Also it is possible to return the wrong
shape from a partition function. We should error in this case.

PiperOrigin-RevId: 525606690

tensorflower-gardener committed:
Add MaybeGetFeature and MaybeGetFeatureValues methods that do not crash

PiperOrigin-RevId: 525608457

tensorflower-gardener committed:
Merge pull request #60305 from VijayR19:master

PiperOrigin-RevId: 525607922

tensorflower-gardener committed:
Rename some metrics to avoid naming violation.

PiperOrigin-RevId: 525608787

tensorflower-gardener committed:
Merge pull request #60349 from npanpaliya:fix-ppc-build

PiperOrigin-RevId: 525610249

Commit:
Add CopyStart and CopyDone in verifier for int4.

PiperOrigin-RevId: 525611065

sagunb committed:
Recursively fold all dependent functions when applying the small_constants_optimization.

PiperOrigin-RevId: 525611375

deqiangc committed:
Check the type of argument before casting it as a tensor type

PiperOrigin-RevId: 525611971

matthiaskramm committed:
Don't set the cluster device if device names are conflicting.

PiperOrigin-RevId: 525613555

wenscarl committed:
Return false if vector bias fusion fails.

anlunx committed:
[xla:gpu] Temporarily disable capturing memcpy to cuda graphs

PiperOrigin-RevId: 525615068

BrianWieder committed:
Expand visibility rules for tensorflow/python/platform/BUILD

PiperOrigin-RevId: 525616016

tensorflower-gardener committed:
Dynamic Device manager list devices in a deterministic order.

After this CL, the device order is determined by ParsedNamed in DeviceNameUtils.

It was in the order of insertion which is nondeterministic, and causes difficulty
in the '2vm' Collective kernel tests that need to workaround Collective V2 kernels
only support PasredNamed ordered list of devices.

PiperOrigin-RevId: 525616108

tensorflower-gardener committed:
Merge pull request #60374 from tensorflow:create-pull-request/patch

PiperOrigin-RevId: 525616541

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525618560

tensorflower-gardener committed:
Export cell_reader to python for test only usage such as benchmarks

PiperOrigin-RevId: 525621710

poulsbo committed:
Move graph_debug_info.proto from tensorflow/core/protobuf to tensorflow/core/framework. This is preparation for further work on stack trace preservation.

PiperOrigin-RevId: 525625142

tensorflower-gardener committed:
Merge pull request #60306 from wenscarl:Fp32_biasAdd_fusion_fix

PiperOrigin-RevId: 525628272

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525637172

tensorflower-gardener committed:
Add a lot of logging to buf_rendezvous.cc

PiperOrigin-RevId: 525641784

tensorflower-gardener committed:
Enable Kernel based collectives between TPU workers.

In our current architecture, The worker service, the eager context must use the same collective manager. The first collective manager, created before CreateContext RPC call, is local. This change replaces that collective manager to an RPC collectife manager using the leader from the RPC request (which is set from remote.py). The pattern only supports a single EagerContext. I think this is almost a fundamental limitation of the EagerService design. I hope I am wrong, and please prove so.

I tried to call UpdateServerDef from the main grpc service, but it deadlocks (likely due to already holding the mutex). It is also unclear to me the relation of the worker cache in grpc service and the one in the worker session created here.

PiperOrigin-RevId: 525645715

poulsbo committed:
Fix breakage in go protos.

PiperOrigin-RevId: 525645956

tensorflower-gardener committed:
#tf-data Support shuffle implementation for UNKNOWN_CARDINALITY constant.

PiperOrigin-RevId: 525648359

dansuh17 committed:
Add missing headers for flatbuffer_export.cc.

This change cleans up the file by adding missing headers for direct usages and removes headers whose imported names are not used.

PiperOrigin-RevId: 525648870

tensorflower-gardener committed:
Support GELU as an elementwise operator in the TFLite GPU delegate.

GELU activations are commonly used in Transformer architectures. The gelu op specification allows for a choice between an approximate and accurate computation.
The accurate version involves the use of erfc, which is not present in OpenGL or Vulkan.

For simplicity, the OpenCL version always uses the accurate form and OpenGL and Vulkan always use the approximate form, ignoring the value set in the op attributes.

PiperOrigin-RevId: 525660188

tensorflower-gardener committed:
Internal Code Change

PiperOrigin-RevId: 525660488

akuegel committed:
[XLA:GPU][NFC] Make CanFoldTransposeIntoDot handle non-canonical dots.

If there is more than one non-contracting dimension, and the layout of the
transpose operand still makes sure that the non-contracting dimensions are
consecutive, it needs to be checked whether the dot shape is still the same after
folding.
Copy several end-2-end tests of folding transposes from gemm_rewrite_test and
make them unit tests, and add one test covering the additional logic.
This change is NFC for now, because we only run TransposeFolding after DotDecomposer, so there currently is always only one non-contracting dimension.

PiperOrigin-RevId: 525661973

akuegel committed:
During GPU layout assignment, don't assume dots have been canonicalized.

This is a first step towards potentially moving DotDecomposer to after layout
assignment.
Also rename some variables to make it clearer what is going on. We were swapping
contracting and non-contracting dimensions twice for the lhs of a dot, so simplify
that to avoid swapping twice.
No impact expected on benchmarks, as we currently do run DotDecomposer before
Layout assignment.

PiperOrigin-RevId: 525672122

sngyhan committed:
Replace CheckNumericsOp to IdentityOp

CheckNumericsOp is not needed for serving.

PiperOrigin-RevId: 525681105

tensorflower-gardener committed:
Update GraphDef version to 1472.

PiperOrigin-RevId: 525683751

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-04-20

PiperOrigin-RevId: 525683761

chhe7 committed:
Makes "mlir emitError()" compatible with absl::string_view for android build.

PiperOrigin-RevId: 525691242

fadara01 committed:
Standardize error messages from invalid shapes for Mkl MatMul and BatchMatMul

- fixes the failing test on Arm CI caused by #60355
- standardizes the use of In[0], In[1] instread of lhs, rhs in error messages

fergushenderson committed:
Fix build breakage on iOS for gpu_plugin-related targets
//tensorflow/lite/experimental/acceleration/configuration:gpu_plugin_impl
//tensorflow/lite/core/acceleration/configuration/c:gpu_plugin
caused by commit 9765df1a06107e3dae851ce38259d25767573058.

There were two issues.  The first one is that we need to ensure that
the gpu_plugin.h header file is built as Objective-C++, not as C++; this is
important because Bazel compiles the header files to verify that they are
self-contained, and this header (conditinally) includes the metal delegate,
which references Objective-C headers, so it won't parse as C++.

The other issue is that it was missing a conditional dependency on the target
for the metal delegate.

PiperOrigin-RevId: 525711555

khasanovaa committed:
Insert PTX inline assembly for vectorization.

PiperOrigin-RevId: 525718007

khasanovaa committed:
Iterate over the different number of chunks and threads.

PiperOrigin-RevId: 525721048

yifjiang committed:
Fix a bug in BFCAllocator::GetFragmentation and the comment for GetGpuHostAllocator

PiperOrigin-RevId: 525724938

khasanovaa committed:
Set parameters to the most efficient value and add __launch_bounds__ hint.

PiperOrigin-RevId: 525729394

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@571eaead173b

Updates LLVM usage to match
[571eaead173b](https://github.com/llvm/llvm-project/commit/571eaead173b)

PiperOrigin-RevId: 525734843

paulinesho committed:
Speed up 4-bit unpacking loop in reference implementation

The new implementation moves the odd-length check out of the main loop, and unrolls slightly differently.

PiperOrigin-RevId: 525759375

arnoegw committed:
Let tf.math.divide_no_nan() accept the same types as tf.math.divide(),
notably integer types, which get promoted to a suitable floating-point
type for the quotient.

This makes it possible to replace any use of divide() by divide_no_nan()
whenever the "no nan" semantics are needed.

PiperOrigin-RevId: 525761523

paynecl committed:
Making a python interface for pulling a list of op names that are compilable for a given backend for documentation purposes.  Backends supported in this API are currently GPU and CPU

PiperOrigin-RevId: 525764676

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525773218

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525773299

olegshyshkov committed:
[XLA:CPU Next] Make inlineFusionCluster work with gml_st.fusion on memrefs.

PiperOrigin-RevId: 525786109

tensorflower-gardener committed:
[xla-next][sparse] fix crash when retrieving constant parameters passed to sparse custom calls.

PiperOrigin-RevId: 525787415

ddunl committed:
Update hlo_test macro

PiperOrigin-RevId: 525789073

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525792724

poulsbo committed:
Add the field debug_info to GraphDef, containing a GraphDebugInfo.
This will be used to represent the stack traces for the nodes of the graph.

PiperOrigin-RevId: 525798472

chuanhaozhuge committed:
Reduce log spam.

PiperOrigin-RevId: 525798797

fionalang committed:
Split `//third_party/tensorflow/python/types:types` into fine-grained, strict targets, and update all references to them.

PiperOrigin-RevId: 525799168

arnoegw committed:
Let tf.nn.compute_average_loss() cope with an empty batch,
even if using the actual batch size from the tensor shape.
This can happen at the end of a distributed dataset.

PiperOrigin-RevId: 525800083

tensorflower-gardener committed:
Hide MaybeCreateNcclCommunicator from callers of Create?CollectiveExecutorMgr.

I had to remove a cycle reference from nccl to cpu_core.

PiperOrigin-RevId: 525800914

rjpower committed:
Adjust checks for `type(Tensor)` to isinstance or is_eager/is_symbolic_tensor.

PiperOrigin-RevId: 525801792

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/5896cea22111836c196493a1898afe4c4fe917ff.

PiperOrigin-RevId: 525802530

tensorflower-gardener committed:
Updates (again) the streamz name to be consistent with others.

PiperOrigin-RevId: 525803154

fionalang committed:
Move `framework/meta_graph._op_name` to `tensor.py` as `tensor.get_op_name` and update the references to it.

This breaks the following cycle:
`ops/resource_variable_ops.py → framework/meta_graph.py → framework/importer.py → framework/function.py → ops/variable_scope.py → ops/variable_v1.py → ops/cond.py → ops/cond_v2.py → ops/default_gradient.py`

PiperOrigin-RevId: 525803207

sirakiin committed:
tf.lite. Add support for i16 to topK

PiperOrigin-RevId: 525806442

fionalang committed:
Split `//third_party/tensorflow/python/types:types` into fine-grained, strict targets, and update all references to them.

PiperOrigin-RevId: 525810953

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525813471

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525813857

fionalang committed:
Split `//third_party/tensorflow/python/types:types` into fine-grained, strict targets, and update all references to them.

PiperOrigin-RevId: 525813980

cantonios committed:
Allow batched encode_png.  The first N-3 dimensions are considered batch
dimensions.

PiperOrigin-RevId: 525816997

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525817434

mpcallanan committed:
Randomize hosts selected to participate in task-level experiments.

Without this change, task 0 is always in the control, which may introduce bias.

PiperOrigin-RevId: 525827133

tensorflower-gardener committed:
[xla][hlo][sparse] Roundtrip new sparse tensor DLT introduced in https://reviews.llvm.org/D148664

PiperOrigin-RevId: 525828395

terryheo committed:
gpu_delegate: Fix a bug when output tensors are shared among nodes

GraphFloat32's outputs is a list of values without consumers.
When a output tensor is consumed by another nodes, it's not listed properly.
This CL resolves it by maintaining `known_graph_outputs_` list.

Also GpuModel has a similar issue and fixed it.

PiperOrigin-RevId: 525831543

tensorflower-gardener committed:
Add support for conditional operator in the auto-sharding pass. Support for sort will soon follow.

PiperOrigin-RevId: 525836165

tensorflower-gardener committed:
[XLA/LayoutAssignment] Add an interface to return a mutable LayoutConstraints pointer.

PiperOrigin-RevId: 525837886

yijie-yang committed:
Move DataExpectation & ShapeExpectation out of TfLiteDriver

PiperOrigin-RevId: 525839165

fionalang committed:
Split `eager/record.py` from `eager/tape.py`.

This is done to prevent import cycles. `framework/ops.py` is a widely used file that uses the logic in `record.py`, so by separating `record.py` from `tape.py`, files which use `tape.py` without needing the `record.py` logic have a lower chance of involvement in an import cycle by not looping in a transitive dependency on `framework/ops.py`.

This change additionally increases modularity in the eager directory.

Also, delete `tape.delete_trace`, which has no usages.

Also fix several build target dependencies that were causing false dependency cycles.

PiperOrigin-RevId: 525840079

nSircombe committed:
Adds mkl_batch_matmul_op to mkl_matmul_op_benchmark build

Following https://github.com/tensorflow/tensorflow/pull/60355, the BUILD
file for mkl_matmul_op_benchmark needs to be updated.

BrianWieder committed:
Create new Windows RBE toolchain

PiperOrigin-RevId: 525847365

hawkinsp committed:
[XLA:Python] Add method to parse HloSharding from its string representation.

PiperOrigin-RevId: 525849193

Commit:
[XLA:GPU] Disable tilings with > 2 pipeline stages on pre-Ampere GPUs in Triton GEMM.

This follows triton/python/triton/ops/matmul_perf_model.py#L154: mma instructions are not used in Triton on pre-Ampere GPUs, also compilation with > 2 stages gets very slow.

PiperOrigin-RevId: 525850327

cota committed:
[xla:cpu:next] fix pre-allocation of descriptor table

PiperOrigin-RevId: 525852835

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525856395

Commit:
[XLA:GPU][NFC] Improve the dynamic shared memory test: invert both rows and columns.

PiperOrigin-RevId: 525858355

swachhandl committed:
Split flag for enabling PjRt for single device compilation into multiple flags

This is so that we can control the rollout across Xla* ops and DeviceTypes.

Users only have control over the op level flags, device allowlists can only be controlled programmatically. If the user level flag is off, the feature will be turned off for all devices.

PiperOrigin-RevId: 525858407

fionalang committed:
Split `//third_party/tensorflow/python/types:types` into fine-grained, strict targets, and update all references to them.

PiperOrigin-RevId: 525859870

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525860908

GleasonK committed:
Integrate StableHLO at openxla/stablehlo@45a85e

PiperOrigin-RevId: 525868856

tensorflower-gardener committed:
Specify platform on API tests.

PiperOrigin-RevId: 525869969

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 525873343

cantonios committed:
Update Eigen to commit:b378014fef017a829fb42c7fad15f3764bfb8ef9

PiperOrigin-RevId: 525874937

tensorflower-gardener committed:
Propagate errors from async checkpoint's async thread to the calller thread.

One of the current issue with TF2 async checkpoint is that errors happening in the async thread are silently ignored as they do not reach the main thread. This CL resolves this issues by propagating the async thread error to the main thread so get the caller's attention.

The implementation stores the async thread error as the internal state in the `async_checkpoint_helper` and emerges it the next time async save/write is called, or when the async save thread is joined.

PiperOrigin-RevId: 525875359

fionalang committed:
Split `//third_party/tensorflow/python/types:types` into fine-grained, strict targets, and update all references to them.

PiperOrigin-RevId: 525876313

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/30df99c1940bdb56ae0fd275c65151e7534c9f15.

PiperOrigin-RevId: 525877339

patnotz committed:
Optimizations for embedding workloads on TPUs.

PiperOrigin-RevId: 525888749

anlunx committed:
[xla:gpu] Add xla.requires_blas attribute to graph capture functions

PiperOrigin-RevId: 525894310

tensorflower-gardener committed:
Add //waymo/ml/cn subpackages to tensorflow internal

PiperOrigin-RevId: 525896225

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@a443b3d18ef4

Updates LLVM usage to match
[a443b3d18ef4](https://github.com/llvm/llvm-project/commit/a443b3d18ef4)

PiperOrigin-RevId: 525898890

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/91e0af3b3b4aefd9580d85e62aaf5e6a92020c76.

PiperOrigin-RevId: 525900913

poulsbo committed:
Update GraphConstructor to take a GraphDebugInfo. This will allow it to reconstruct the stack trace of the nodes in the graph.

PiperOrigin-RevId: 525909857

tensorflower-gardener committed:
Op documentation update.
	update of g3doc/_includes/tf_passes.md
	update of g3doc/hlo_ops.md

PiperOrigin-RevId: 525911784

tensorflower-gardener committed:
Update auto assignment list

PiperOrigin-RevId: 525929431

sngyhan committed:
Support weight-only quantization in the TPU inference converter

This adds another quantization scheme that quantizes only weights
and leave activation in float.

PiperOrigin-RevId: 525960113

tensorflower-gardener committed:
Update GraphDef version to 1473.

PiperOrigin-RevId: 525973002

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-04-21

PiperOrigin-RevId: 525973000

jblespiau committed:
Simplify the code to use `absl::StatusCode`.

PiperOrigin-RevId: 525979450

Commit:
[XLA:GPU] Add API to query full CUDA shared memory size including the dynamic one; use it for Triton-based GEMM.

PiperOrigin-RevId: 525983143

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@9e9b1effac34

Updates LLVM usage to match
[9e9b1effac34](https://github.com/llvm/llvm-project/commit/9e9b1effac34)

PiperOrigin-RevId: 525989002

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/7042a7854b798ded049fa9f35b5b193da445d95b.

PiperOrigin-RevId: 525990559

Commit:
Remove dependency on shims folder

Replace includes of tensorflow/lite/core/shims/ headers to tensorflow/lite/.
Replace dependencies on tensorflow/lite/core/shims: targets with tensorflow/lite:.

PiperOrigin-RevId: 526013381

tensorflower-gardener committed:
Add use-case property to GPU compatibility list.

PiperOrigin-RevId: 526018909

olegshyshkov committed:
[XLA:CPU Next] Add bufferizable op interface for gml_st.fusion.

gml_st.fusion bufferizes as dst-style op. It easier this way, because we fusion clusters consist of other dst-stype ops and we construct clusters in a way, that we already know appropriate init buffers and read/write pattern for arguments.

PiperOrigin-RevId: 526035675

tensorflower-gardener committed:
Automated visibility attribute cleanup.

PiperOrigin-RevId: 526038353

tensorflower-gardener committed:
Merge pull request #60370 from elfringham:clean_arm_ci_extended

PiperOrigin-RevId: 526041221

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@e46fcabe84ec

Updates LLVM usage to match
[e46fcabe84ec](https://github.com/llvm/llvm-project/commit/e46fcabe84ec)

PiperOrigin-RevId: 526050283

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/9b9b36f11c4822da78869c6fdfbee1a6b68db985.

PiperOrigin-RevId: 526053423

fionalang committed:
Split `//third_party/tensorflow/python/types:types` into fine-grained, strict targets, and update all references to them.

PiperOrigin-RevId: 526063161

Maratyszcza committed:
[XNNPack] Relax constraints on delegated MEAN nodes

PiperOrigin-RevId: 526065173

tensorflower-gardener committed:
Merge pull request #60390 from nSircombe:bugfix/matmul_bm_build

PiperOrigin-RevId: 526067475

glowzbd committed:
Do not print out the detail logs when enable_logging is true.

PiperOrigin-RevId: 526072134

matthiaskramm committed:
Mark TF_TPUAnnotateTensorsWithDynamicShapeOp as Pure.

PiperOrigin-RevId: 526074324

tensorflower-gardener committed:
Merge pull request #60360 from Intel-tensorflow:bhavanis/onednn-3.0-mklquantize

PiperOrigin-RevId: 526077824

tensorflower-gardener committed:
Stop caching list of devices in WorkerEnv

This change to prepare for 'dynamic' virtual device support in tensorflow single controller
system's workers, which will make DTensor and outside compilation in general
easier on these workers.

In that case, the worker's virtual device list can be mutated via a WorkerService RPC call (see comment in cl/525889023), making it more consistent with the Python execution path. Then the list is no longer static and the caching gets into the way.

Even in the current status, it is unclear which of the users actually benefit from having the cache. (none of the users appear to be on a performance critical path).

I am also renaming a few StaticDeviceMgr to DynamicDeviceMgr, as the former has been converted to an alias of the later earlier of the day.

PiperOrigin-RevId: 526082955

tensorflower-gardener committed:
Merge pull request #60240 from ROCmSoftwarePlatform:fix_cwise_ops_test_import

PiperOrigin-RevId: 526086716

tensorflower-gardener committed:
Add an overloaded method Interpreter::SetBufferHandle(...) with param TfLiteTensor*

PiperOrigin-RevId: 526098270

jurahul committed:
Revert "Automated visibility attribute cleanup."

PiperOrigin-RevId: 526100539

tensorflower-gardener committed:
[XLA/Conditional code motion] Do not move operand inside conditional branches if the operand is also used in operand 0 of the conditional.

PiperOrigin-RevId: 526103598

nmjohn committed:
Add AddV2 to flops_registry.

PiperOrigin-RevId: 526103883

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 526107474

jurahul committed:
[Revert] Add an overloaded method Interpreter::SetBufferHandle(...) with param TfLiteTensor*

PiperOrigin-RevId: 526111779

tensorflower-gardener committed:
Avoid automated removal of visibility.

PiperOrigin-RevId: 526115397

faizan-m committed:
Add FunctionType to AtomicFunction

PiperOrigin-RevId: 526122057

BrianWieder committed:
Delete `python/util/compat_internal` since it is not used anywhere anymore.

PiperOrigin-RevId: 526123582

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@fd2211d84a07

Updates LLVM usage to match
[fd2211d84a07](https://github.com/llvm/llvm-project/commit/fd2211d84a07)

PiperOrigin-RevId: 526124843

matthiaskramm committed:
Add op definition for tf.ResourceGatherNd.

PiperOrigin-RevId: 526125664

tensorflower-gardener committed:
Merge pull request #60388 from fadara01:standardize_mkl_matmul_error_messages

PiperOrigin-RevId: 526125294

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/b65453e2449fbedccedade4ae886c0226c1583f6.

PiperOrigin-RevId: 526127068

mpcallanan committed:
Make `OwnedIterator` exportable to saved model.

PiperOrigin-RevId: 526127362

changm committed:
Create a test for xla_scatter op in preparation to legalize it in bridge phase 2.

PiperOrigin-RevId: 526127938

Maratyszcza committed:
Revert "Relax constraints on delegated MEAN nodes"

PiperOrigin-RevId: 526129361

sparc1998 committed:
In future cls, we want to be able to construct a HeapSimulator::Chunk from an offset and end. So, create 2 factory constructors, FromOffsetSize() and FromOffsetEnd(). Switch all current cases to use the FromOffsetSize().

Also, add a Chunk::ToString() method for debugging.

PiperOrigin-RevId: 526134581

cantonios committed:
Add ApproxTopK kernels for CPU/GPU for inputs compatible with TopK.

Simply register the original TopK kernel.  This is to support some models that
use ApproxTopK with default settings.

PiperOrigin-RevId: 526138858

SandSnip3r committed:
Change run_hlo_module to deduce the input type from the file extension, when no input type is given.

PiperOrigin-RevId: 526140811

SeeForTwo committed:
Remove full type information when optimizing a node to a NoOp

Graph rewrites that change a node's op to a NoOp should clear any full type information from the node (since NoOps have no output and full type information describes the output of the node). This CL fixes graph rewrites in core/grappler/optimizers/constant_folding.cc, core/grappler/optimizers/arithmetic_optimizer.cc and core/tpu/graph_rewrite/combine_tpu_embedding_load_retrieve_pass.cc.

Create ChangeToNoOp helper function in core/framework/node_def_util.*. Because of cases where an optimization pass that replaces a node with NoOp is followed by another pass that deletes the node, independently testing that full type is correct for each code path is challenging. Using the common helper function eases achieving good test coverage.

PiperOrigin-RevId: 526141222

lrdxgm committed:
Add more IWYU pragmas to export definitions from internal headers, and to mark internal headers private.

PiperOrigin-RevId: 526146851

hhb committed:
Cleanup: More RefCountPtr for Rendezvous

PiperOrigin-RevId: 526149895

tensorflower-gardener committed:
[Revert Revert] Add an overloaded method Interpreter::SetBufferHandle(...) with param TfLiteTensor*

PiperOrigin-RevId: 526150966

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 526152829

tensorflower-gardener committed:
Update cache key for TF graph optimization.

PiperOrigin-RevId: 526158458

tensorflower-gardener committed:
This is the first of several CLs implementing automatic mesh selection for the auto-sharding pass. This CL adds a method to create multiple mesh shapes to try as part of this effort.

PiperOrigin-RevId: 526161397

tensorflower-gardener committed:
Add support for sort operator in the auto-sharding pass. We merely enumerate all possible shardings for the op. The resharding costs are already correctly handled by relying on the sharding propagation pass. We further compute non-zero communication costs when the op is sharded across the sort dimension. This is currently done by approximating the costs as the same as AllToAll costs, though that can be improved/changed.

PiperOrigin-RevId: 526163758

fionalang committed:
Remove `framework/func_graph.py`'s dependency on `autograph`.

Move the part of the implementation of `func_graph.func_graph_from_py_func` that relies on `autograph` to `eager/polymorphic_function/tracing_compiler.py`, the only place where it is used. This introduces a direct dependency from `tracing_compiler.py` onto `autograph`, which was there transitively before.

This breaks several cycles, most importantly:
framework:func_graph -> autograph:autograph -> autograph/operators:\_\_init\_\_ -> autograph/operators:control_flow -> framework:func_graph

Also, add several missing build dependencies exposed by this refactor.

PiperOrigin-RevId: 526167883

SandSnip3r committed:
Add --force_fake_data to run_hlo_module. The flag only has an effect when the input is an hlo snapshot.

PiperOrigin-RevId: 526169212

pkanwar23 committed:
[XLA:GPU] Add API to query full CUDA shared memory size including the dynamic one; use it for Triton-based GEMM.

PiperOrigin-RevId: 526170961

fionalang committed:
Delete deprecated `//third_party/tensorflow/python/types:types` target now that usages have been migrated to the fine-grained targets.

PiperOrigin-RevId: 526175112

tensorflower-gardener committed:
Add ApproxTopK kernels for CPU/GPU for inputs compatible with TopK.

Simply register the original TopK kernel.  This is to support some models that
use ApproxTopK with default settings.

PiperOrigin-RevId: 526176719

wilsingosti committed:
#tf-data Ramp down both `autotune_buffer_optimization` and `stage_based_autotune_v2` experiments to fix a bug.

PiperOrigin-RevId: 526180096

tensorflower-gardener committed:
Add specialized GPU TopK kernel.

This faster version works for K<16 and relies on keeping a sorted list of values
in registers to avoid data movement as much as possible.

The algorithm is essentially broken down into two sections:
 1. Slice the data equally betwen warps and compute the warp-local topk.
	- First we split the data into equal-sized chunks, load and sort the first K
		elements in registers.
	- Next each thread iterates over the remainder of its data, keeping the
		in-register array sorted.
	- Finally, on each warp we reduce the lane-local topk to a warp-local topk.
	- One thread for each warp writes the warp-local topk to shared memory.
 2. On a single warp:
 	- Lane i reads the warp-local topk from warp i.
	- Now that each lane has the corresponding warp-local topk, compute the global
		topk.

For efficiency, we pack float32 & bf16 into u64/u32 togehter with their indices.
This promotes better codegen and cheaper comparisons.

PiperOrigin-RevId: 526185694

tensorflower-gardener committed:
Shift some runtime_shape.h definitions to .cc to reduce binary sizes

PiperOrigin-RevId: 526186778

Commit:
[XLA] Fix latency hiding scheduler delay send-done logic.

The send done logic tries to delay send-dones to avoid triggering them too early.
This logic was broken, because the ready time of the send() attached to the send-done()
is not actually populated as it is not ready yet. We need to estimate as the ready time
of the successors + latency.

This was causing the delay logic to always triggering making effectively kScheduleDone policy
never trigger for send-dones() host.

PiperOrigin-RevId: 526200252

mikedelorimier committed:
Remove `enable_logging` from TPU bridge functions. No callers were using this. If a test needs to turn on dumping it can set TF_DUMP_GRAPH_NAME_FILTER instead.

PiperOrigin-RevId: 526209124

tensorflower-gardener committed:
Internal Code Change

PiperOrigin-RevId: 526219215

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@8db947da0b37

Updates LLVM usage to match
[8db947da0b37](https://github.com/llvm/llvm-project/commit/8db947da0b37)

PiperOrigin-RevId: 526235855

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/eda65a38acebfa4b107c42bff1d33cec712440bf.

PiperOrigin-RevId: 526237308

tensorflower-gardener committed:
Update GraphDef version to 1474.

PiperOrigin-RevId: 526247027

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-04-22

PiperOrigin-RevId: 526247074

tensorflower-gardener committed:
Generalize the small topk kernel to arbitrary aligned N.

PiperOrigin-RevId: 526350414

tensorflower-gardener committed:
Hide MaybeCreateNcclCommunicator from callers of Create?CollectiveExecutorMgr.

I had to remove a cycle reference from nccl to cpu_core.

PiperOrigin-RevId: 526364089

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-04-23

PiperOrigin-RevId: 526399496

tensorflower-gardener committed:
Update GraphDef version to 1475.

PiperOrigin-RevId: 526399497

chaokunyang committed:
fix xla operation semantics alltoall typo

jakeharmon8 committed:
Pass api_gen outputs via parameter file

PiperOrigin-RevId: 526498877

tensorflower-gardener committed:
Make globals safe against at-exit data race

PiperOrigin-RevId: 526518332

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/9a87d04a0b586c2c7ef792c9c57a010cd531c5e7.

PiperOrigin-RevId: 526530236

Commit:
Change from std::vector::at() to std::vector::operator[] for retrieving OpKernel for performance.

The potential bugs for retrieving OpKernel should be caught during unit tests.

PiperOrigin-RevId: 526549222

tensorflower-gardener committed:
Update GraphDef version to 1476.

PiperOrigin-RevId: 526574135

tensorflower-gardener committed:
compat: Update forward compatibility horizon to 2023-04-24

PiperOrigin-RevId: 526574143

Commit:
[XLA:GPU][NFC] Remove a flaky autotuning test.

The test is not reliable anymore with many more enabled configurations. SelectsSplitK above it implements the initial purpose better: making sure that the autotuner selects configurations based on the measured execution time.

PiperOrigin-RevId: 526582429

Commit:
[XLA:GPU] Add API to query full CUDA shared memory size including the dynamic one; use it for Triton-based GEMM.

PiperOrigin-RevId: 526599252

tensorflower-gardener committed:
Merge pull request #60293 from ashay:ashay/fix-cmake-build

PiperOrigin-RevId: 526605942

tensorflower-gardener committed:
Add plain addition model test in MiniBenchmark C API.

This change adds test which runs Minibenchmark on plain addition model and ensures that the produced benchmark output is correct.

It also fixes a subtle bug where the usage of `reserve` instead of `resize` string method results in `absl::StrCat` not able to correctly read the string returned from `GenerateRandomString` call, thereby always concatenating an empty string instead of random one.

PiperOrigin-RevId: 526627346

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 526641665

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 526648756

fionalang committed:
Turn import of `eager/wrap_function` in `framework/convert_to_constants` into a regular import.

PiperOrigin-RevId: 526669330

jurahul committed:
[NFC][XLA/GPU] Add a pass to annotate async collectives with sync/async flag
- The end goal here is to only have the async forms of HLO instructions
  in the XLA/GPU backend.
- This will use a new GPU specific backend config.

PiperOrigin-RevId: 526670868

tensorflower-gardener committed:
Merge pull request #60402 from chaokunyang:fix_operation_semantics_typo

PiperOrigin-RevId: 526671618

hehhjiang committed:
Recalculate iterator position when serializing multiple BenchmarkEvent to bytes.

PiperOrigin-RevId: 526674072

tensorflower-gardener committed:
Integrate LLVM at llvm/llvm-project@ba38640b9901

Updates LLVM usage to match
[ba38640b9901](https://github.com/llvm/llvm-project/commit/ba38640b9901)

PiperOrigin-RevId: 526679978

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/61a2d6bdbac6bcf9fdd31b3fffa736727fa7fa10.

PiperOrigin-RevId: 526684421

wilsingosti committed:
#tf-data Some model nodes are not optimized in stage based autotune and autotune buffer optimization algorithms because the `collect_node` predicate passed into `CollectNodesLocked()` function does not visit the whole sub tree of a node if the `collect_node(node)` returns false. This CL fixes this bug.

PiperOrigin-RevId: 526685748

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/04ed615cbbaa26e3add7868dfedf8277700e79a4.

PiperOrigin-RevId: 526687457

Commit:
Add an overload of BEFFunction::Execute() to pass arguments by moving the reference, so that we don't need to use extra reference count in while kernel implementation.

PiperOrigin-RevId: 526688364

tensorflower-gardener committed:
Update TFRT dependency to use revision
http://github.com/tensorflow/runtime/commit/7d879c8b161085a4374ea481b93a52adb19c0529.

PiperOrigin-RevId: 526691835

jurahul committed:
[XLA/GPU] Add GPU specific async -> sync collective conversion pass
- The GPU specific variant will use the CollectiveBackendConfig to
  annotate async form of collectives in HLO to convert them to sync.
- We still use the analysis from the generic form of the pass that
  converts HLO async collectives to HLO sync collectives.

PiperOrigin-RevId: 526698565

changm committed:
Legalize XlaSetDynamicDimensionSize with tf2xla kernel

PiperOrigin-RevId: 526700891

BrianWieder committed:
Update references to `tensorflow/python:platform` to the new fine-grained targets.

PiperOrigin-RevId: 526703802

BrianWieder committed:
Add missing dependencies for tensorflow/python/platform:self_check, tensorflow/python/platform:test, and tensorflow/python/platform:app.

PiperOrigin-RevId: 526708386

tensorflower-gardener committed:
Fix a typo, and handle a resulting, previously unhandled case when generating input shardings for Gather ops in order to compute resharding costs during auto-sharding.

PiperOrigin-RevId: 526708587

BlaziusMaximus committed:
Make fingerprint exception log more explicit.

PiperOrigin-RevId: 526712765

ianstenbit committed:
Fix use of `reshape` in TF Numpy API implementation of `tile`

PiperOrigin-RevId: 526713447

kenfranko committed:
Pass frontend attributes to XLA host send/recv callbacks.

These attributes are used in Tensorflow to generate the Rendezvous keys
used for outside compilation.

PiperOrigin-RevId: 526716260

wenchenvincent committed:
Merge (unresolved conflicts) remote-tracking branch 'upstream/master' into develop-upstream-sync-230424

wenchenvincent committed:
Merge (resolved conflicts) remote-tracking branch 'upstream/master' into develop-upstream-sync-230424

jayfurmanek commented on tensorflow/compiler/xla/service/gpu/tests/BUILD:
```diff
@@ -812,10 +812,6 @@ xla_cc_binary(
         "//tensorflow/tsl/platform:rocm_rocdl_path",
         "//tensorflow/compiler/xla/service/gpu:amdgpu_compiler_impl",
-<<<<<<< HEAD
-	"//tensorflow/tsl/platform:rocm_rocdl_path",
```
This was added in: https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/commit/dfb7dd0b2f7e6c6bbcd3f1dbc9e299be9554b4a7

At the time, it was needed. Curious if that is still the case.

cota committed:
legalize_mhlo: add missing llvm:: namespace to cast ops

PiperOrigin-RevId: 527080438

fionalang committed:
Pull variable creation logic out of `ops/variable_scope.py`.

Since `variable_scope.py` is currently responsible for both variable scoping and much of the variable creation logic, it has dependencies on `variable_v1.py` and `ref_variable.py` that are causing cycles. This change involves pulling the variable creation logic and proto registrations out of `variable_scope.py` and into `ref_variable.py` and `resource_variable_ops.py` (for tf1 and tf2 respectively). This is part one of two changes to remove `variable_scope`'s explicit dependency on variable_v1 logic, which breaks many import cycles.

A lazy load in `ref_variable.py` is needed while the refactors are in progress.

PiperOrigin-RevId: 527488606

wenchenvincent committed:
Disabled the following tests for weekly sync 230424:

//tensorflow/compiler/xla/service/gpu/tests:copy_nested_amdgcn.hlo.test
//tensorflow/compiler/xla/service/gpu/tests:fused_scatter_amdgcn.hlo.test
//tensorflow/compiler/xla/service/gpu/tests:launch_dimensions_amdgcn.hlo.test

wenchenvincent committed:
Disabled a check in //tensorflow/python/framework:function_test_cpu

The history of disabling this check for ROCm dates back to the following
commit:
https://github.com/tensorflow/tensorflow/commit/a5f86f3f01329914aaf7e9238262608046ee3aa8#diff-afbffd4d8f4108b0c3c7e8e800eae061385894fd880e22037d3613cb01f6c23d

jayfurmanek commented:
It looks like we need this one now too:
https://github.com/tensorflow/tensorflow/commit/e6aa56fad3d7d24561d9e7321a4b6b991b74762f

Since estimator got ahead of us with:
https://github.com/tensorflow/estimator/commit/ad72355f9f766fda7bf5cdedd9146b8d787458db

wenchenvincent commented:
> It looks like we need this one now too: [tensorflow@e6aa56f](https://github.com/tensorflow/tensorflow/commit/e6aa56fad3d7d24561d9e7321a4b6b991b74762f)
> 
> Since estimator got ahead of us with: [tensorflow/estimator@ad72355](https://github.com/tensorflow/estimator/commit/ad72355f9f766fda7bf5cdedd9146b8d787458db)

Is this commit to fix the following error?
AttributeError: module 'tensorflow.python.ops.ref_variable' has no attribute '_to_proto_fn'

Now all the CI test failures are due to this error. But when I tried to run them on a local MI60, there were no failures. So it seems that there is some difference between the docker container used by the CI and rocm/tensorflow-autobuilds:latest that led to the discrepancy.

jayfurmanek commented:
> Is this commit to fix the following error? AttributeError: module 'tensorflow.python.ops.ref_variable' has no attribute '_to_proto_fn'

yes

> Now all the CI test failures are due to this error. But when I tried to run them on a local MI60, there were no failures. So it >seems that there is some difference between the docker container used by the CI and rocm/tensorflow-autobuilds:latest that >led to the discrepancy.

yeah, the first commit above went in to TF last week. The second one followed in TFE yesterday and wound up in `tf-estimator-nightly` May 01 build. The "Ubuntu-GPU-*" jobs build the old container each time and pull in the latest `tf-estimator-nightly` each time which is why were are seeing it there. The "gpu-non-pip' jobs pull and use the tensorflow-build container which get published on each new push to the repo, so those don't yet have the tf-estimator-nightly May 01 build.

Every once in a while there are some interdependencies like this and we got bit here since this sync is a little late. I just did a test build locally (with tf-estimator-ngihtly May01) and this commit does indeed fix the issue. I'll push the cherry-pick.
